sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -85,12 +85,24 @@ case class DecimalType(precision: Int, scale: Int) extends FractionalType {
   private[sql] def isWiderThan(other: DataType): Boolean = other match {
     case dt: DecimalType =>
       (precision - scale) >= (dt.precision - dt.scale) && scale >= dt.scale
-    case dt: IntegralType =>
```

Comment:
That is correct.  They could always truncate so we should return this and fix the function below.